packages/sdk/src/reliable_channel/missing_message_retriever.ts

```diff
@@ -0,0 +1,78 @@
+import type {
+  IDecodedMessage,
+  IDecoder,
+  QueryRequestParams
+} from "@waku/interfaces";
+import type { MessageId } from "@waku/sds";
+import { Logger } from "@waku/utils";
+
+const log = new Logger("sdk:missing-message-retriever");
+
+const DEFAULT_RETRIEVE_FREQUENCY_MS = 10 * 1000; // 10 seconds
+
+export class MissingMessageRetriever<T extends IDecodedMessage> {
+  private retrieveInterval: ReturnType<typeof setInterval> | undefined;
+  private missingMessages: Map<MessageId, Uint8Array<ArrayBufferLike>>; // Waku Message Ids
+
+  public constructor(
+    private readonly decoder: IDecoder<T>,
+    private readonly retrieveFrequencyMs: number = DEFAULT_RETRIEVE_FREQUENCY_MS,
+    private readonly _retrieve: <T extends IDecodedMessage>(
+      decoders: IDecoder<T>[],
+      options?: Partial<QueryRequestParams>
+    ) => AsyncGenerator<Promise<T | undefined>[]>,
+    private readonly onMessageRetrieved?: (message: T) => Promise<void>
+  ) {
+    this.missingMessages = new Map();
+  }
+
+  public start(): void {
+    if (this.retrieveInterval) {
```

Comment:
since you reset interval on consecutive start should `missingMessages` be reset too?

Reply:
The messages are still missing in the channel. We just ensure that the interval doesnt get corrupted.